PR for Release 1.4 (#300)
nf-core/rnaseq: CHANGELOG.md

# nf-core/rnaseq: Changelog

## Version 1.4dev

### Pipeline updates

* Add option `--skipTrimming` to be able to run from previous processed data since this step take quite long.

> * Restore `SummarizedExperimment` object creation in the salmon_merge process avoiding increaasing memory with sampe size.

Reviewer: ```suggestion
* Restore `SummarizedExperiment` object creation in the `salmon_merge` process to avoid increasing memory with sample size.
```

* Adjust wording of skipped samples [in pipeline output](https://github.com/nf-core/rnaseq/issues/290)
* Added [possibility to output unmapped reads](https://github.com/nf-core/rnaseq/issues/277) to pipeline
* Add option to modify `featureCounts -t` option to assign reads to a different type than `exon`. This makes it compatible with nuclearRNAseq. For instance, it would count reads in introns and exons if `gene` is used instead. Nuclear RNA has un-spliced RNA, and the whole transcript, including the introns, need to be counted, e.g. by specifying `--fc_count_type transcript`.
* Fix sample names in feature counts and dupRadar to remove suffixes added in other processes
* Removed `genebody_coverage` process [#195](https://github.com/nf-core/rnaseq/issues/195)
* Implemented Pearsons correlation instead of euclidean distance [#146](https://github.com/nf-core/rnaseq/issues/146)
* Add `--stringTieIgnoreGTF` parameter [#206](https://github.com/nf-core/rnaseq/issues/206)
* Resolved link to guidelines is broken [#203](https://github.com/nf-core/rnaseq/issues/203)
* Removed unnecessary `stringtie` channels for `MultiQC`
* Added tximport to merge salmon output

> * Added Salmon as an supplementary method to STAR and HiSAT2

Reviewer: We could do with tidying up this changelog quite a bit - maybe move some of these to the start, with a summary of the major changes first?

Author: Summary added, added some headings, too to get this a bit easier to follow and read. Let me know what you think, added all of this to `https://github.com/nf-core/rnaseq/tree/mini-release-fixes`, so that others in the team can adjust / help on this too 👍

* Added `--psuedo_aligner`, `--transcript_fasta` and `--salmon_index` parameters
* Add `Citation` and `Quick Start` section to `README.md`
* Closed missing multiqc_plots in dev branch output [#200](https://github.com/nf-core/rnaseq/issues/200)
* Integrate changes in `nf-core/tools v1.6` template which resolved [#90](https://github.com/nf-core/rnaseq/issues/90)

> * Moved process "convertGFFtoGTF" before "makeSTARindex" [#215](https://github.com/nf-core/rnaseq/issues/215)

Reviewer: ```suggestion
* Moved process `convertGFFtoGTF` before `makeSTARindex` [#215](https://github.com/nf-core/rnaseq/issues/215)
```

* Add tximport and summarizedexperiment dependency [#171](https://github.com/nf-core/rnaseq/issues/171)
* Change all boolean parameters from snake_case to camelCase and vice versa for value parameters

> * Appointed changes because of missing output of the multiqc_plots folder [#200](https://github.com/nf-core/rnaseq/issues/200)

Reviewer: ```suggestion
* Fixed MultiQC config to correctly export `multiqc_plots` folder [#200](https://github.com/nf-core/rnaseq/issues/200)
```

* Add Qualimap dependency [#202](https://github.com/nf-core/rnaseq/issues/202)
* Add SM ReadGroup info for QualiMap compatibility[#238](https://github.com/nf-core/rnaseq/issues/238)
* Obtain edgeR + dupRadar version information [#198](https://github.com/nf-core/rnaseq/issues/198) and [#112](https://github.com/nf-core/rnaseq/issues/112)
* Get MultiQC to save plots as [standalone files](https://github.com/nf-core/rnaseq/issues/183)

> * Get MultiQC to save plots as [standalone files](https://github.com/nf-core/rnaseq/issues/183): added the folder "multiqc_plots" to the output.

Reviewer: ```suggestion
```

> * Get MultiQC to write out the software versions in a .csv file [#185](https://github.com/nf-core/rnaseq/issues/185)

Reviewer: ```suggestion
* Get MultiQC to write out the software versions in a `.csv` file [#185](https://github.com/nf-core/rnaseq/issues/185)
```

# nf-core/rnaseq: Changelog

## Version 1.4

Major novel changes include:

* Support for Salmon as an alternative method to STAR and HISAT2
* Several improvements in `featureCounts` handling of types other than `exon`. It is possible now to handle nuclearRNAseq data. Nuclear RNA has un-spliced RNA, and the whole transcript, including the introns, needs to be counted, e.g. by specifying `--fc_count_type transcript`.
* Support for [outputting unaligned data](https://github.com/nf-core/rnaseq/issues/277) to results folders.
* Added options to skip several steps

  * Skip trimming using `--skipTrimming`
  * Skip BiotypeQC using `--skipBiotypeQC`

>   * Skip Alignment using `-skipAlignment` to only use pseudo-alignment using Salmon

Author: Mhm, could be a good addition I agree 🤔

### Documentation updates

* Adjust wording of skipped samples [in pipeline output](https://github.com/nf-core/rnaseq/issues/290)
* Fixed link to guidelines [#203](https://github.com/nf-core/rnaseq/issues/203)
* Add `Citation` and `Quick Start` section to `README.md`
* Add in Documentation of the `--gff` parameter

### Reporting Updates

* Generate MultiQC plots in the results directory [#200](https://github.com/nf-core/rnaseq/issues/200)
* Get MultiQC to save plots as [standalone files](https://github.com/nf-core/rnaseq/issues/183)
* Get MultiQC to write out the software versions in a `.csv` file [#185](https://github.com/nf-core/rnaseq/issues/185)
* Use `file` instead of `new File` to create `pipeline_report.{html,txt}` files, and properly create subfolders

### Pipeline enhancements & fixes

> * Restore `SummarizedExperimment` object creation in the salmon_merge process avoiding increaasing memory with sample size.

Author: I guess this means 1.) here but @lpantano might be the person to ask that 👍
